imports/api/users.js

```diff
@@ -47,11 +47,16 @@ if (Meteor.isServer) {
         // delete data.email;
 
         // save oldEmail address
-        const oldEmail = Meteor.users.findOne(this.userId).emails[0].address
-        // remove old email address
-        Accounts.removeEmail(this.userId, oldEmail)
+        const user = Meteor.users.findOne(this.userId)
+        const oldEmail = user.emails && user.emails[0] && user.emails[0].address
+
         // add new email address
         Accounts.addEmail(this.userId, data.email, false)
+
+        if (oldEmail) {
```

Comment:
This change fixed the following bug:

Steps:
- user attempts to edit email
- enters an email that is already taken by another user

Expected:
- user sees an error message and has to retry

Actual:
- no error message, and user's email is deleted and leaves the account with no emails